### PR TITLE
Automate the markdown generation for the CSS variables.

### PR DESCRIPTION
**Pull request by petyosi**
Needs more work. 

Currently, running `yarn build-docs` will output an HTML table with some variables parsed. I believe we should write those to several files and have them included in the markdown. Here's what it looks like:

<table>
      <thead><tr><th>Name</th><th>Default Value</th><th>Description</th><th>Used in</th></tr></thead>
      <tbody>
          <tr><td><code>--str-chat__primary-color</code></td><td>var(--str-chat__blue500)</td><td>Used for emphasis, brands can inject their main color using this variable</td><td>Used in Links, buttons, avatar fallback background, etc. </td></tr>
<tr><td><code>--str-chat__primary-color-low-emphasis</code></td><td>var(--str-chat__blue300)</td><td>Used for emphasis, brands can inject their main color using this variable, it has less emphasis than primary color</td><td>Used in Selected message reaction background color </td></tr>
<tr><td><code>--str-chat__active-primary-color</code></td><td>var(--str-chat__blue600)</td><td>Used to indicate that a UI element with primary color is in an active state</td><td>Used in Pressed buttons </td></tr>
<tr><td><code>--str-chat__on-primary-color</code></td><td>var(--str-chat__grey50)</td><td>If the primary color is used as a background, text/icons are displayed in this color</td><td>Used in Button color, text on avatar fallback, etc. </td></tr>
<tr><td><code>--str-chat__background-color</code></td><td>var(--str-chat__grey50)</td><td>Used as a background color for the main chat UI components</td><td>Used in Channel list background </td></tr>
<tr><td><code>--str-chat__secondary-background-color</code></td><td>var(--str-chat__grey50)</td><td>Used as a background color for the main chat UI components</td><td>Used in Channel background, message list background, message input background </td></tr>
<tr><td><code>--str-chat__primary-surface-color</code></td><td>var(--str-chat__blue100)</td><td>Used as a background color to give emphasis, but less vibrant than the primary color</td><td>Used in Background of messages sent by the current user </td></tr>
<tr><td><code>--str-chat__surface-color</code></td><td>var(--str-chat__grey300)</td><td>A neutral color used to give emphasis to different surfaces</td><td>Used in Selected channel preview background, borders </td></tr>
<tr><td><code>--str-chat__secondary-surface-color</code></td><td>var(--str-chat__grey200)</td><td>A neutral color used to give emphasis to different surfaces</td><td>Used in Background of received messages, reactions, etc. </td></tr>
<tr><td><code>--str-chat__text-color</code></td><td>var(--str-chat__grey950)</td><td>The main color used for texts/icons</td><td>Used in Messages, channel header, channel title in the channel preview, etc. </td></tr>
<tr><td><code>--str-chat__text-low-emphasis-color</code></td><td>var(--str-chat__grey500)</td><td>Used for texts/icons that need less emphasis</td><td>Used in The latest message in the channel preview, deleted message text, default icon state, etc. </td></tr>
<tr><td><code>--str-chat__disabled-color</code></td><td>var(--str-chat__grey400)</td><td>Used for displaying disabled UI elements (typically buttons)</td><td>Used in Disabled send button etc. </td></tr>
<tr><td><code>--str-chat__on-disabled-color</code></td><td>var(--str-chat__grey50)</td><td>Used for text/icon colors if disabled color is used as a background color</td><td>Used in Text on disabled send button etc. </td></tr>
<tr><td><code>--str-chat__danger-color</code></td><td>var(--str-chat__red400)</td><td>Used for error messages, and destructive actions</td><td>Used in Delete action label etc. </td></tr>
<tr><td><code>--str-chat__online-indicator-color</code></td><td>var(--str-chat__green500)</td><td></td><td></td></tr>
<tr><td><code>--str-chat__message-highlight-color</code></td><td>var(--str-chat__yellow100)</td><td></td><td></td></tr>
<tr><td><code>--str-chat__unread-badge-color</code></td><td>var(--str-chat__red400)</td><td>Used for displaying the unread badge</td><td>Used in Background of the unread badge </td></tr>
<tr><td><code>--str-chat__on-unread-badge-color</code></td><td>var(--str-chat__grey50)</td><td>Used for text/icon colors if unread badge color is used as a background color</td><td>Used in Text on the unread badge </td></tr>
<tr><td><code>--str-chat__overlay-color</code></td><td>rgba(252, 252, 252, 0.9)</td><td>The background color used for overlays</td><td>Used in Modal overlay color, attachment preview overlay </td></tr>
<tr><td><code>--str-chat__secondary-overlay</code></td><td>rgba(0, 0, 0, 0.2)</td><td>The background color used for subtle overlays</td><td>Used in Image gallery </td></tr>
<tr><td><code>--str-chat__secondary-overlay-text-color</code></td><td>var(--str-chat__grey50)</td><td>The text/icon color used on subtle overlays</td><td>Used in Image gallery </td></tr>
      </tbody>
    </table>

### CODE REVIEW

**Review by oliverlaz — APPROVED** [review]

File: scripts/generate-variables.mjs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+              if (comment.usage) {
+                variableInfo.usage = comment.usage;
+              }
+
+              if (selector === LIGHT_THEME_SELECTOR) {
+                processedCssVariables.lightTheme.push(variableInfo);
+                return;
+              }
+
+              if (selector === DARK_THEME_SELECTOR) {
```
Comment (szuperaz) [inline scripts/generate-variables.mjs:82]: We sometimes use the dark theme selector inside components: https://github.com/GetStream/stream-chat-css/blob/theming2/src/v2/styles/MessageActionsBox/MessageActionsBox-theme.scss#L28 - I think that would categorize the `--str-chat__message-actions-box-box-shadow` with the theme variables

Comment (szuperaz) [reply scripts/generate-variables.mjs:82]: But that's an issue with the CSS code, we probably shouldn't have the dark selector inside a component file, maybe we should create a theme variable for box shadow?

**Review by szuperaz — APPROVED** [review]

File: scripts/generate-variables.mjs (hunk 1 above)
Comment (szuperaz) [reply scripts/generate-variables.mjs:82]: Done

**Comment by szuperaz** [comment]
Other PR: https://github.com/GetStream/stream-chat-css/pull/143

**Pull request closed without merging**